### PR TITLE
feat(lint): Adds automated JS linting and style checking to the build…

### PR DESCRIPTION
**Pull request by bryanbraun**
… process.

This includes the following tools:
- Using jshint for linting, with rules defined in .jshintrc
- Using jscs for code style checks, with rules defined in .jscsrc
- Using jsdoc for inline function documentation, with rules defined at
  the bottom of .jscsrc

The specific rules are subject to discussion and change. They were
simply lifted from one of our client projects.

This linting and style checks are applied to all files that match `src/js/**/*.js`

The build is designed to fail if any of this linting errors out.

### CODE REVIEW

File: tasks/test.coffee [hunk 1]
```diff
@@ -54,4 +56,8 @@ module.exports = (gulp, cfg, env) ->
 
       checkStatus()
 
-  gulp.task "test", ['tests-integration', 'tests-unit']
+  gulp.task 'jsstyle-tests', ->
+    gulp.src(['!specs/integration/lib/*.js', 'specs/**/*.js'])
+      .pipe(jscs())
+
+  gulp.task "test", ['jsstyle-tests', 'tests-integration', 'tests-unit']
```
Comment (cromwellryan) [inline tasks/test.coffee]: Looks like `jshint` task is missing?

**Comment by asimpson** [comment]
we should also add `src/js` to `watch`. It was originally set up by some bozo(me) for Coffeescript.

File: tasks/test.coffee (hunk 1 above)
Comment (bryanbraun) [inline tasks/test.coffee]: Good catch.

**Comment by bryanbraun** [comment]
@cromwellryan, @asimpson, I made those updates.

**Comment by asimpson** [comment]
:+1: 

getinsertpic.com

**Pull request closed without merging**